GW2EIEvtcParser/EncounterLogic/Fractals/SilentSurf/Kanaxai.cs

```diff
@@ -297,6 +301,117 @@ internal override void ComputePlayerCombatReplayActors(AbstractPlayer player, Pa
+            }
+
+            // Numbers spread AoEs
+            IEnumerable<Segment> spreads = player.GetBuffStatus(log, KanaxaiSpreadOrangeAoEBuff, log.FightData.LogStart, log.FightData.LogEnd).Where(x => x.Value > 0);
+            foreach (Segment spreadSegment in spreads)
+            {
+                replay.Decorations.Add(new CircleDecoration(true, 0, 380, ((int)spreadSegment.Start, (int)spreadSegment.End), "rgba(200, 120, 0, 0.2)", new AgentConnector(player)));
+            }
+        }
+
+        internal override void ComputeEnvironmentCombatReplayDecorations(ParsedEvtcLog log)
+        {
+            base.ComputeEnvironmentCombatReplayDecorations(log);
+
+            // Red AoE during Frightening Speed attack
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.FrighteningSpeedRedAoE, out IReadOnlyList<EffectEvent> frighteningSpeedRedAoEs))
+            {
+                foreach (EffectEvent aoe in frighteningSpeedRedAoEs)
+                {
+                    int duration = 1500;
+                    int start = (int)aoe.Time;
+                    int effectEnd = start + duration;
+                    EnvironmentDecorations.Add(new CircleDecoration(true, 0, 380, (start, effectEnd), "rgba(255, 0, 0, 0.2)", new PositionConnector(aoe.Position)));
+                    EnvironmentDecorations.Add(new CircleDecoration(false, 0, 380, (start, effectEnd), "rgba(255, 0, 0, 0.2)", new PositionConnector(aoe.Position), 10));
+                }
+            }
+
+            // Axe AoE
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.AxeGroundAoE, out IReadOnlyList<EffectEvent> axeAoEs))
+            {
+                foreach (EffectEvent aoe in axeAoEs)
+                {
+                    // Get World Cleaver casts and find the first cast time event present after the AoE effect time
+                    AgentItem agentKanaxai = log.AgentData.GetAgentByType(AgentItem.AgentType.NPC).FirstOrDefault(x => x.ID == (int)ArcDPSEnums.TargetID.KanaxaiScytheOfHouseAurkusCM);
+                    var kanaxai = new NPC(agentKanaxai);
+                    IReadOnlyList<AbstractCastEvent> casts = kanaxai.GetCastEvents(log, log.FightData.FightStart, log.FightData.FightEnd);
+                    AbstractCastEvent cast = casts.Where(x => x.SkillId == WorldCleaver).FirstOrDefault(x => x.Time > aoe.Time);
+                    long worldCleaverTime = cast != null ? cast.Time : 0;
+
+                    // Get Axe AoE Buffs
+                    IReadOnlyList<AbstractBuffEvent> buffs69195 = log.CombatData.GetBuffData(RendingStormAxeTargetBuff1);
+                    IReadOnlyList<AbstractBuffEvent> buffs69208 = log.CombatData.GetBuffData(RendingStormAxeTargetBuff2);
+                    var axes = new List<AbstractBuffEvent>();
+                    axes.AddRange(buffs69208);
+                    axes.AddRange(buffs69195);
+                    var orderedAxes = axes.OrderBy(x => x.Time).ToList();
```

Comment:
Same, this can be put outside of the for loop.

Reply:
Is there a better way to merge the two lists and ordering the final one than my 6 steps?

Reply:
You don't need to create variables for 69195 and 69208, you can call the APIs directly within axes.AddRange, which will shove off 2 steps.